src/GatherContent/HTTPClient.php

```diff
@@ -63,4 +73,65 @@ private function sendRequest($url, $additionalCurlOptions)
         return $response;
     }
 
+    private function sendMultiRequest($urls, $additionalCurlOptions)
+    {
+        // data to be returned
+        $results = array();
+
+        // array of curl handles
+        $channels = array();
+
+        // initialize the multihandler
+        $mh = curl_multi_init();
+
+        $active = 0;
+        $urlIndex = 0;
+
+        // Process Loop
+        do {
+            // Add to the in flight requests
+            $numToAdd = $this->totalConcurrentRequests - $active;
+
+            while ($numToAdd-- && $urlIndex < count($urls)) {
```

Comment:
I think this do...while construct, in addition to "$numToAdd--" in the inner while conditional, could cause an infinite loops if the functions is called with $urls set to an empty array.

Reply:
If $urls is empty, $urlIndex < count($urls) is false so it should break out of this on both this loop and the outer do...while loop, right?

Reply:
and what happens on line 110?